src/Exthost/LanguageInfo.rei

```diff
@@ -12,12 +12,12 @@ let toString: t => string;
 
 let defaultLanguage: string;
 
-let getLanguageFromExtension: (t, string) => string;
 let getLanguageFromFilePath: (t, string) => string;
 let getLanguageFromBuffer: (t, Buffer.t) => string;
 
 let getScopeFromLanguage: (t, string) => option(string);
-let getScopeFromExtension: (t, string) => option(string);
+let getScopeFromFileName: (t, string) => option(string);
+let getScopeFromBuffer: (t, Buffer.t) => option(string);
```

Comment:
Dropped the extension based functions from this interface file, as they aren't used and using just the extension gives us less info that the filename.

The functions still exist or are easy to add so if we do end up needing it, its no issue, but I think having them accessible will mean that they get used more readily when the filename itself gives us 3 ways of determining the language vs 1 way with the extension.

Reply:
Cool! Ya, good to get rid of these - less to think about, and it's likely that using them won't give the complete picture with the new logic 👍